aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePluginConfiguration.java

```diff
@@ -31,39 +30,22 @@
 final class AWSDataStorePluginConfiguration {
 
     private final SyncMode syncMode;
-    private final String apiName;
 
-    private AWSDataStorePluginConfiguration(final SyncMode syncMode, final String apiName) {
+    private AWSDataStorePluginConfiguration(final SyncMode syncMode) {
         this.syncMode = syncMode;
-        this.apiName = apiName;
     }
 
     static AWSDataStorePluginConfiguration fromJson(JSONObject pluginJson) throws DataStoreException {
         // If no sync mode is specified, we just use the default (no sync) and continue
-        if (!pluginJson.has("syncMode")) {
-            return new AWSDataStorePluginConfiguration(SyncMode.LOCAL_ONLY, null);
+        if (pluginJson == null || !pluginJson.has("syncMode")) {
```

Comment:
If pluginJson is empty, would `has` return false and not throw any exception?

Reply:
if pluginJson is null, it would throw NPE :)

Reply:
The idea is to now allow null objects to be passed as pluginConfiguration parameter to prevent crashes from happening on category level